packages/azure-kusto-ingest/test/e2eTests/e2eTest.ts

```diff
@@ -25,17 +25,21 @@ import {
     StreamingIngestClient,
     IngestionStatus,
     IngestionResult,
-} from "../../src";
-import { sleep } from "../../src/retry";
+} from "azure-kusto-ingest";
+
+import { sleep } from "../../src/retry.js";
 
 import { AzureCliCredential } from "@azure/identity";
 import assert from "assert";
 import fs, { ReadStream } from "fs";
 import util from "util";
 import { v4 as uuidv4 } from "uuid";
-import pathlib from "path";
+import { basename, dirname } from "path";
 import sinon from "sinon";
-import { TableReportIngestionResult } from "../../src/ingestionResult";
+import { fileURLToPath } from 'node:url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
```

Comment:
__dirname is not supported officially in esm.

In node 20.11 they have import.meta.dirname to replace it, but this way is supported from 10+